simple_ldap_user/simple_ldap_user.module

```diff
@@ -480,16 +480,21 @@ function simple_ldap_user_sync_user_to_drupal($drupal_user) {
   // Load the LDAP user, force a cache reset.
   $ldap_user = SimpleLdapUser::singleton($drupal_user->name, TRUE);
 
-  // Nothing to sync.
   if(isset($drupal_user->uid) && $drupal_user->uid == 1) {
     // Check if user is an admin.
+    // Nothing to sync.
     return;
-  } elseif (!$ldap_user->exists) {	//For other users.
+  } 
+  else {
+    $status = 1;	//to unblock users
+    if(!$ldap_user->exists) { 
+      $status = 0;	//to block users
+    }
```

Comment:
There's no need to run an update if the user is already active.  You have the `$drupal_user` object already loaded.  Check it to see if the update is needed.